emcc.py

```diff
@@ -1990,8 +1991,13 @@ def compile_source_file(i, input_file):
     link_flags = [l for l in link_flags if l[1] not in ('-lm', '-lrt', '-ldl', '-lpthread')]
     temp_files = filter_out_dynamic_libs(temp_files)
 
-    # Decide what we will link
-    if not (shared.Settings.WASM_BACKEND and shared.Settings.WASM_OBJECT_FILES):
+    link_to_object = final_suffix not in executable_endings
+    using_lld = shared.Settings.WASM_BACKEND and not (link_to_object and not shared.Settings.WASM_OBJECT_FILES)
```

Comment:
This might be clearer if the `not` was distributed. It's also confusing that `using_lld` is false when `link_to_object` is true. Surely we still use lld when linking to create objects? Could this be resolved with different names?

Reply:
We do use lld for linking to object files.  But for linking of bitcode into other bitcode we still use llvm-link.   Does that make sense?

Reply:
Basically for the wasm backend we always use lld unless we are trying to produce a bitcode file.  lld cant' do that.